assets/js/admin/admin.js

```diff
@@ -1,13 +1,48 @@
 const { __, sprintf } = wp.i18n;
+const schemify = require( 'url-schemify' );
 
 jQuery( document ).ready( function( $ ) {
 	const $form = $( '#post' );
+	const $urlFields = $( '.cmb2-text-url' );
 	const $toValidate = $( '[data-validation]' );
 
+	$urlFields.blur( ( e ) => {
```

Comment:
@jhung Latest commit adds a function to ensure that entered URLs have a scheme when the field is blurred. To test, add a URL without http or https, then click out of the field.

Reply:
@greatislander this causes another issue. If you add a link but don't provide an URL, you must manually "X" the text field. Otherwise upon submission there will be an error displayed ("http://" is submitted).

Another way we can handle this would be add "http://" to the string after submission if it is a valid archival URL, but lacks the scheme.

In the case where just the scheme is submitted, we can either silently ignore it (treat it as blank), or throw an error. I prefer the former as it's likely the intent of the user not to specify any information.

Reply:
Update: I see now if the URL is valid but lacks the scheme, a scheme is added.

Given this behaviour, we would still have to handle the case where the user clicks "Add link", leaves the URL blank, and submits.

Reply:
I've updated this functionality—the scheme will only be added if there's text in the field (see 675f3c7).